src/Aspire.Hosting/Dashboard/DashboardServiceData.cs

```diff
@@ -50,31 +49,9 @@ static GenericResourceSnapshot CreateResourceSnapshot(IResource resource, string
                     State = snapshot.State?.Text,
                     StateStyle = snapshot.State?.Style,
                     HealthStatus = snapshot.HealthStatus,
-                    HealthReports = GetOrCreateHealthReports(),
+                    HealthReports = snapshot.HealthReports,
                     Commands = snapshot.Commands
                 };
-
-                ImmutableArray<HealthReportSnapshot> GetOrCreateHealthReports()
```

Comment:
What is the reason for removing this?

I think we want to see health reports for a resource even if it isn't running yet. The health reports are still in a running state. The health reports would show a waiting status. Is this lost?

Alternatively, is it no longer needed because the snapshot always contains the health reports for health check annotations?

Reply:
Done now in `GetInitialHealthReports`?

Reply:
Yep